packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -42,6 +54,8 @@ inline jobject create_java_exception(JNIEnv *jenv, realm_error_t error) {
                                               error_type_class,
                                               "asThrowable",
                                               "(IILjava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)Ljava/lang/Throwable;", true);
+
+    jenv->PushLocalFrame(3);
```

Comment:
How's this approach different from calling `DeleteLocalRef` on each local reference? 
also we need to check for OOM for each `PushLocalFrame` call

Reply:
Added OOM checks.